cli/deployments/traefik.go

```diff
@@ -45,8 +47,32 @@ func (k Traefik) Describe() string {
 	return emoji.Sprintf(":cloud:Traefik version: %s\n:clipboard:Traefik Ingress chart: %s", traefikVersion, traefikChartURL)
 }
 
+// Delete removes traefik from kubernetes cluster
 func (k Traefik) Delete(c kubernetes.Cluster) error {
-	return c.Kubectl.CoreV1().Namespaces().Delete(context.Background(), traefikDeploymentID, metav1.DeleteOptions{})
+	message := "Deleting Traefik"
+	s := spinner.New(spinner.CharSets[11], 100*time.Millisecond)
+	s.Suffix = emoji.Sprintf(" %s :zzz:", message)
+	s.Start()
+
+	currentdir, err := os.Getwd()
+	if err != nil {
+		return errors.New("Failed uninstalling Traefik: " + err.Error())
+	}
+
+	helmCmd := fmt.Sprintf("helm uninstall traefik --namespace '%s'", traefikDeploymentID)
+	if out, err := helpers.RunProc(helmCmd, currentdir, k.Debug); err != nil {
+		return errors.New("Failed uninstalling Traefik: " + out)
+	}
+
+	err = c.Kubectl.CoreV1().Namespaces().Delete(context.Background(), traefikDeploymentID, metav1.DeleteOptions{})
+	if err != nil {
+		return errors.New("Failed uninstalling Traefik: " + err.Error())
+	}
+	s.Stop()
```

Comment:
Move up to `s.Start()` and use `defer` to ensure execution regardless of code path.

Reply:
Done with commit 84de7f0c9c6094ee525ad5b8033ae42f11225d03.